slides/04_quality.qmd

```diff
@@ -457,21 +447,61 @@ R package [testthat](https://cran.r-project.org/package=testthat)
 
 ```{r, echo = TRUE}
 library(testthat)
-object <- getMean(c(1, 2, 3))
-expected <- 2
-expect_equal(object, expected)
+expect_equal(getMean(c(1, 3, 2)), 2)
 ```
 
 **Example**: unit test failed
 
 ```{r, echo = TRUE, eval = FALSE}
-expect_equal(getMean(1:4), 2)
+expect_equal(getMean(c(1, 3, 2, NA)), 2)
+expect_equal(getMedian(c(1, 3, 2)), 2)
 ```
 
 ::: red-small
-Error: mean(1:4) (actual) not equal to 2 (expected).
+Error: getMean(c(1, 3, 2, NA)) not equal to 2.
+Error: getMedian(c(1, 3, 2)) not equal to 2.
```

Comment:
this seems to be an actual mistake in the code in earlier versions (?). Using this as a teaching moment about continually writing tests.